returnn/search.py

```diff
@@ -185,6 +191,66 @@ def run(self):
             out.write("}\n")
 
 
+class SearchWordsToCTM(Job):
+    """
+    Convert RETURNN search output file into CTM format file
+    """
+
+    def __init__(self, recog_words_file, corpus, filter_tags=True):
+        """
+        :param Path recog_words_file: search output file from RETURNN
+        :param Path corpus: bliss xml corpus
+        :param bool filter_tags: if set to True, tags such as [noise] will be filtered out
+        """
+        self.recog_words_file = recog_words_file
+        self.corpus = corpus
+        self.filter_tags = filter_tags
+
+        self.out_ctm_file = self.output_path("search.ctm")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        corpus = Corpus()
+        corpus.load(self.corpus.get_path())
+        d = eval(open(self.recog_words_file.get_path(), "r").read())
+        assert isinstance(
+            d, dict
+        ), "only search output file with dict format is supported"
+        with open(self.out_ctm_file.get_path(), "w") as out:
+            out.write(
+                ";; <name> <track> <start> <duration> <word> <confidence> [<n-best>]\n"
+            )
+            for seg in corpus.segments():
+                seg_start = 0.0 if seg.start == float("inf") else seg.start
+                seg_end = 0.0 if seg.end == float("inf") else seg.end
+                seg_fullname = seg.fullname()
+                assert seg_fullname in d, "can not find {} in search output".format(
+                    seg_fullname
+                )
+                out.write(";; %s (%f-%f)\n" % (seg_fullname, seg_start, seg_end))
+                words = d[seg_fullname].split()
+                # Just linearly interpolate the start/end of each word as time stamps are not given
+                avg_dur = (seg_end - seg_start) * 0.9 / max(len(words), 1)
```

Comment:
this will be zero if a corpus has e.g. `seg.end == float("inf")`, is that okay?

Reply:
should be ok. i think i had such a case. also the time stamps are not used. or maybe there is better solution not sure.